src/main/java/com/laserfiche/repository/api/clients/impl/RepositoriesClientImpl.java

```diff
@@ -48,6 +51,52 @@ public RepositoryInfo[] getRepositoryList() {
                 throw ApiClientUtils.createApiException(httpResponse, problemDetails);
             }
         };
-        return ApiClientUtils.sendRequestWithRetry(httpClient, httpRequestHandler, baseUrl + "/v1/Repositories", "GET", null, null, null, null, null, null, new HashMap<String, String>(), false, parseResponse);
+        return ApiClientUtils.sendRequestWithRetry(httpClient, httpRequestHandler, baseUrl + "/v1/Repositories", "GET",
+                null, null, null, null, null, null, new HashMap<String, String>(), false, parseResponse);
+    }
+
+    /**
+     * Returns the repository resource list that current user has access to given the API server base URL. Only available in Laserfiche Self-Hosted.
+     *
+     * @param url API server base URL e.g., https://{APIServerName}/LFRepositoryAPI
+     * @return Get the respository resource list successfully.
+     */
+    public static RepositoryInfo[] getSelfHostedRepositoryList(String url) {
```

Comment:
should this follow the same pattern as the other api functions with a ParametersFor function argument?

Reply:
i just decided to avoid wrapping it in an encapsulation object and keep it simple like all the `nextlink` functions since it is only 1 parameter